docparser/model.go

```diff
@@ -374,14 +374,23 @@ func replaceSchemaNameToCustom(s *schema) {
 	if len(refSplit) != 4 {
 		return
 	}
+
 	if replacementSchema, found := registeredSchemas[refSplit[3]]; found {
-		meta, ok := replacementSchema.(metaSchema)
+		meta, ok := replacementSchema.(*schema)
 		if !ok {
 			return
 		}
-		refSplit[3] = meta.CustomName()
+		switch meta.Type {
```

Comment:
So, this implem works, but I actually found where the ACTUAL issue is. This is more of a hack.

It works with `struct` thanks to this:
https://github.com/alexjomin/openapi-parser/blob/01dbd97d5e83eadd8573509a3278bdda96c5007b/docparser/model.go#L571

But this is not done for `map` or `array`. Everything should happen in:
https://github.com/alexjomin/openapi-parser/blob/01dbd97d5e83eadd8573509a3278bdda96c5007b/docparser/model.go#L520

Reply:
I'm guessing this would put the name of the alias as $ref which is what it did before the "break" so that's good.

Though I was considering going further and replacing it altogether by the target of the alias.

Reply:
I think that may require a bit of a rework, if I'm not mistaken. But I wouldn't mind the change, if you're up for it.

Reply:
What do you think @denouche?

Reply:
@denouche Do you have time to review this one?

Reply:
Hey, sorry for the delay. I just took the time to read this thread.
If I understand well the debate is what to do between:
- going further and replacing it by the target of the alias
- or just fixing the initial issue
?

Reply:
Sorry for the delay as well, don't know how I missed your reply.

I think it could be done iteratively, first avoiding the issue and then going further but I would have to retest with new changes first.